core/src/main/java/com/linecorp/armeria/server/annotation/Cookies.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.annotation;
+
+import java.util.Set;
+
+import io.netty.handler.codec.http.cookie.Cookie;
+
+/**
+ * An interface which holds decoded {@link Cookie} instances for an HTTP request.
+ */
+public interface Cookies extends Set<Cookie> {
+
+    /**
+     * Creates an instance with the specified set of {@link Cookie}s.
+     */
+    static Cookies from(Set<Cookie> cookies) {
```

Comment:
How about:

- Adding:
  - `static Cookies of(Cookie... cookies)`
  - `static Cookies copyOf(Iterable<? extends Cookie> cookies)`
  - .. which make a copy of the given collection
- and removing `from()` and calling `new DefaultCookies()` directly internally, so we do not expose a potentially unsafe operation to our users?

Reply:
> and removing from() and calling new DefaultCookies() directly internally, so we do not expose a potentially unsafe operation to our users?

Replaced `Cookies.from(...)` with `Cookies.copyOf(...)`. Because `DefaultCookies` is package-private, so it should be at the same package with `Cookies` interface. Because `ImmutableSet` is passed into `Cookies.copyOf(...)`, there might not perform actual copy operation.